extensions/src/main/java/org/pentaho/platform/plugin/services/importer/PentahoPlatformImporter.java

```diff
@@ -183,6 +185,44 @@ public static String computeBundlePath( String bundlePath ) {
     return bundlePath;
   }
 
+  /**
+   * Performs one-way conversion on incoming String to produce a syntactically valid JCR path (section 4.6 Path Syntax).
+   */
+  public static String checkAndSanitize( final String in ) {
+    if ( in == null ) {
+      throw new IllegalArgumentException();
+    }
+    String extension = null;
+    if ( in.endsWith( RepositoryObjectType.CLUSTER_SCHEMA.getExtension() ) ) {
```

Comment:
This should probably be a switch case instead of an if / else stack. Then the default case can throw a illegalargumentexception.